test/integration/causal-consistency/causal_consistency.prose.test.js

```diff
@@ -0,0 +1,285 @@
+              expect(secondFindCommand.readConcern).to.have.any.key('afterClusterTime');
+              expect(secondFindCommand.readConcern.afterClusterTime).to.eql(firstOperationTime);
+            });
+        }
+      });
+    }
+  );
+
+  it(
+    '6. A read operation in a ClientSession that is not causally consistent should not include the afterClusterTime parameter in the command sent to the server',
+    /**
+     * session = client.startSession(causalConsistency = false)
+     * collection.anyReadOperation(session, {})
+     * operationTime = session.operationTime
+     * capture the command sent to the server (using APM or other mechanism)
+     * assert that the command does not have an afterClusterTime field
+     */
+    {
+      metadata: {
+        requires: { topology: ['replicaset'], mongodb: '>3.6.0-rc0' },
+        // Skipping session leak tests b/c these are explicit sessions
+        sessions: { skipLeakTests: true }
+      },
+
+      test: function () {
+        const session = test.client.startSession({ causalConsistency: false });
+        const db = test.client.db(this.configuration.db);
+        const coll = db.collection('causal_test', { readConcern: { level: 'majority' } });
+
+        return coll
+          .findOne({}, { session: session })
+          .then(() => coll.findOne({}, { session: session }))
+          .then(() => {
+            const commands = test.commands.started.map(command => command.command);
+            expect(commands).to.have.length(2);
+            for (const command of commands) {
+              expect(command).to.have.any.key('readConcern');
+              expect(command.readConcern).to.not.have.any.key('afterClusterTime');
+            }
+          });
+      }
+    }
+  );
+
+  it(
+    '7. A read operation in a causally consistent session against a deployment that does not support cluster times does not include the afterClusterTime parameter in the command sent to the server',
+    /**
+     * session = client.startSession(causalConsistency = true)
+     * collection.anyReadOperation(session, {})
+     * capture the command sent to the server (using APM or other mechanism)
+     * assert that the command does not have an afterClusterTime field
+     */
+    {
+      metadata: { requires: { topology: ['single'], mongodb: '>3.6.0-rc0' } },
+
+      test: function () {
+        const db = test.client.db(this.configuration.db);
+        const coll = db.collection('causal_test', { readConcern: { level: 'local' } });
+
+        return coll
+          .findOne({})
+          .then(() => coll.findOne({}))
+          .then(() => {
+            const command = test.commands.started[1].command;
+            expect(command).to.have.any.key('readConcern');
+            expect(command.readConcern).to.not.have.any.key('afterClusterTime');
+          });
+      }
+    }
+  );
+
+  // TODO[CHECK WITH TEAM]: This may be an invalid spec test case in light of SPEC-1019
```

Comment:
Open question here, kind of a confusing situation since it seems that SPEC-1019 has removed this from being relevant, not sure if there's a drivers ticket to update this case or if it's just an oversight